### DISCUSSION TITLE
gcc 12.2.0 failing to build on Mac M1

### DISCUSSION THREAD

**Discussion by filbranden** [Homebrew/discussions#3701] [Everyday usage] [answered]
I have a Mac M1 running Monterey 12.5.1 and Homebrew installed on a custom prefix `~/.homebrew`

Upgrade gcc 12.1.0 to 12.2.0 has been failing with:

```
==> ../configure --prefix=/Users/filbranden/.homebrew/opt/gcc --libdir=/Users/filbranden/.homebrew/opt/gcc/lib/gcc/current --disable-nls --enable-checking=release --with-gc
==> make
Last 15 lines from /Users/filbranden/Library/Logs/Homebrew/gcc/02.make:
7  0x1a15981b4  _dispatch_client_callout + 20
8  0x1a15a9a04  _dispatch_root_queue_drain + 680
9  0x1a15aa104  _dispatch_worker_thread2 + 164
10  0x1a1758324  _pthread_wqthread + 228
A linker snapshot was created at:
	/tmp/libstdc++.6.dylib-2022-09-26-061803.ld-snapshot
ld: Assertion failed: (_file->_atomsArrayCount == computedAtomCount && "more atoms allocated than expected"), function parse, file macho_relocatable_file.cpp, line 2061.
collect2: error: ld returned 1 exit status
make[6]: *** [libstdc++.la] Error 1
make[5]: *** [all-recursive] Error 1
make[4]: *** [all-recursive] Error 1
make[3]: *** [all] Error 2
make[2]: *** [all-stage1-target-libstdc++-v3] Error 2
make[1]: *** [stage1-bubble] Error 2
make: *** [all] Error 2

Do not report this issue to Homebrew/brew or Homebrew/core!

These open issues may also help:
clip: drop GCC dependency https://github.com/Homebrew/homebrew-core/pull/110763
djview4: drop GCC dependency https://github.com/Homebrew/homebrew-core/pull/110664
openmsx: drop GCC dependency https://github.com/Homebrew/homebrew-core/pull/110324
simutrans: drop GCC dependency https://github.com/Homebrew/homebrew-core/pull/110143
Remove Linux-only GCC dependencies https://github.com/Homebrew/homebrew-core/issues/110010
synergy-core: drop GCC dependency https://github.com/Homebrew/homebrew-core/pull/110128
gcc@11: do not strip on macOS https://github.com/Homebrew/homebrew-core/pull/110706
GCC 12.2.0 breaks TLS (thread_local) at least on x86_64 https://github.com/Homebrew/homebrew-core/issues/110673
```

The open issues don't seem to be relevant to this problem here.

It seems the problem is not necessarily with gcc 12.2.0 itself, but perhaps with using gcc 12.1.0 to *build* gcc 12.2.0 itself.

I found a post related to this issue here: https://developer.apple.com/forums/thread/714565

But Apple support in that post essentially just points out that that's building using Homebrew compilers and linkers and not Xcode, so it's not something they're going to look into fixing. (They actually suggested starting a discussion here.)

I imagine the custom prefix is what's triggering the rebuild in my case (a bottle might have worked fine), but wondering how I can work around this issue. (Perhaps force using Xcode to build Homebrew gcc would avoid this issue? I imagine this might be the environment in which the bottle gets built? Is there a way to enforce that even when a Homebrew gcc is already installed?)

Thanks!

**Comment by filbranden** [comment]
Uploading the full log file: [02.make](https://github.com/Homebrew/discussions/files/9658087/02.make.txt)

**Comment by Bo98** [comment, marked as answer]
This is an Xcode 14 `ld` bug. The only workaround is to downgrade to Xcode 13. Apple are aware and are working on a fix.

I haven't tested 14.1 beta 2 yet - it might already be fixed there.

**Reply by filbranden** [reply]
Downgrading to "Command Line Tools for Xcode 13.4" did indeed fix this issue. Thanks!
